src/rpc/blockchain.cpp

```diff
@@ -670,6 +670,32 @@ static bool GetUTXOStats(CCoinsView *view, CCoinsStats &stats)
     return true;
 }
 
+UniValue pruneblockchain(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() != 1)
+        throw runtime_error(
+            "pruneblockchain\n"
+            "\nArguments:\n"
+            "1. \"height\"       (int, required) The block height to prune up to.\n");
+
+    if (pruneMode != PRUNE_MANUAL) {
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Cannot prune via RPC unless in manual prune mode.");
+        return 0;
+    }
+
+    LOCK(cs_main);
+
+    int height = params[0].get_int();
+    if (height < 0) {
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Negative block height.");
+    } else if ((unsigned int) chainActive.Height() < Params().PruneAfterHeight()) {
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Blockchain is too short for pruning.");
+    }
```

Comment:
Should this also check if the requested height is too close to the tip? Or maybe just return the height it was able to successfully prune up to...